### PR TITLE
Sidebar updates

### PR DESCRIPTION
**Pull request by kellyjandrews**
This pull request modifies the sidebar navigation in several ways.

1.  **Side bar categories are now collapsable.** 

2. **Side bar categories are context aware**

3. **Sidebar categories are now controlled in the [`_config.yml` file](https://github.com/codeship/documentation/compare/master...sidebar-updates?expand=1#diff-aeb42283af8ef8e9da40ededd3ae2ab2R90)** 
4. **Categories have moved to [`_config.yml` defaults ](https://github.com/codeship/documentation/compare/master...sidebar-updates?expand=1#diff-aeb42283af8ef8e9da40ededd3ae2ab2R25)**
5. **Pro/Basic sidebar includes General categories**

### CODE REVIEW

**Comment by kellyjandrews** [comment]
Ah yeah - I forgot I saw the small width issue. I can go back and clean it up. 

Not sure I would be able to split things up into 2 PRs here without manual effort.  I can try to make the PRs smaller in the future.

File: _config.yml [hunk 1]
```diff
@@ -86,40 +165,40 @@ kramdown:
     line_numbers: false
 
 # Prose.io
```
Comment (mlocher) [inline _config.yml]: Need not happen in this PR, but I think we can throw out Prose for now. If we want to add it again, we'd need to start from scratch anyways.

File: _config.yml [hunk 2]
```diff
@@ -22,6 +22,104 @@ defaults:
       weight: 50
       comments: true
       image: /assets/img/og_documentation.jpg
+  - scope:
```
Comment (mlocher) [inline _config.yml]: I'm personally not a fan of having the category defined outside of the YAML frontmatter. I know that with the directory structure we have right now it's easy to do this, but I prefer to have it assigned explicitly on the post itself.

Comment (ethangj) [inline _config.yml]: If the folder structure is always available, I'm not sure it makes a big difference, but if we could end up in a situation where we want to categorize separate from the folder structure then we should probably keep front matter in place.

**Comment by ethangj** [comment]
@kellyjandrews @mlocher 

I'd go ahead and merge this. I created new cards assigned to Kelly to create category overviews and reintroduce frontmatter, and put a card on to investigate menus for a longer-term solution.

**Pull request merged**